src/lib.rs

```diff
@@ -37,7 +40,7 @@ enum WatcherEnum {
 
 #[pyclass]
 struct RustNotify {
-    changes: Arc<Mutex<HashSet<(u8, String)>>>,
+    changes: Arc<Mutex<HashSet<(u8, String, String)>>>,
```

Comment:
```suggestion
    changes: Arc<Mutex<HashSet<(u8, Option<String>, String)>>>,
```

maybe? Then you can use `None` for the `src` for `added`, `moved` and `deleted`?

Reply:
Yes I tried that before, and run into issues that I didn't really understand.